public/static/app/signal-processing/SignalProcessingCtrl.js

```diff
@@ -1,28 +1,87 @@
 mciModule.controller('SignalProcessingCtrl', function(
-  $window, Stitch, STITCH_CONFIG
+  $window, $scope, Stitch, STITCH_CONFIG
 ) {
   var vm = this;
   var projectId = $window.project
 
-  vm.isLoading = true
-  Stitch.use(STITCH_CONFIG.PERF).query(function(db) {
-    return db
-      .db(STITCH_CONFIG.PERF.DB_PERF)
-      .collection(STITCH_CONFIG.PERF.COLL_CHANGE_POINTS)
-      .find({})
-      .limit(100)
-      .execute()
-  }).then(function(docs) {
-    vm.gridOptions.data = docs
-  }, function(err) {
-    console.error(err)
-  }).finally(function() {
-    vm.isLoading = false
-  })
+  var state = {
+    sorting: null,
+    filtering: {}
+  }
+
+  function loadData(state) {
+    vm.isLoading = true
+    Stitch.use(STITCH_CONFIG.PERF).query(function(db) {
+      return db
+        .db(STITCH_CONFIG.PERF.DB_PERF)
+        .collection(STITCH_CONFIG.PERF.COLL_CHANGE_POINTS)
+        .aggregate(getAggChain(state))
+    }).then(function(docs) {
+      vm.gridOptions.data = docs
+    }, function(err) {
+      console.error(err)
+    }).finally(function() {
+      vm.isLoading = false
+    })
+  }
+
+  loadData(state)
+
+  function getAggChain(state) {
+    var chain = []
+
+    // Desctruct.
+    var sorting = state.sorting
+    var filtering = state.filtering
+
+    if (!_.isEmpty(filtering)) {
+      chain.push(mdbFiltering(filtering))
+    }
+
+    if (sorting) {
+      chain.push(mdbSorting(sorting))
+    }
+
+    chain.push({$limit: 250})
```

Comment:
Fixing this is a bit non-trivial; Going to create a ticket for infinite scroll. For now, I believe PERF engineers won't scrolling 250 lines of change points. I am sure filtering/sorting will help to deal with this issue.

Reply:
We will be using a view and grouping by revision and probably project (this brought 5602 change points down to 213).

I would hold off implementing infinite scroll in the short term (and increase the limit to 500).